### PR TITLE
chore: update @hashicorp/platform-analytics and @types/react to v18

### PR DESCRIPTION
**Pull request by BRKalow**
- Install `@hashicorp/platform-analytics@latest @hashicorp/platform-types@latest`
- Update react types to v18 so that we can pull in updated `@hashicorp/react-` packages

### CODE REVIEW

File: src/components/bookmark-button/toast/make-bookmark-toast.tsx [hunk 1]
```diff
@@ -14,7 +15,7 @@ type ToastActionConfig = Record<
 	{
 		title: string
 		descriptionSuffix: string
-		icon: JSX.IntrinsicElements['svg']
```
Comment (BRKalow) [inline src/components/bookmark-button/toast/make-bookmark-toast.tsx:17]: This is an interface that represents the only props of the `svg` element, not a component accepts the same props as `svg`. I've updated all instances to be `ReactElement<React.JSX.IntrinsicElements['svg']>`

File: src/components/button-link/index.tsx [hunk 2]
```diff
@@ -57,9 +57,11 @@ const ButtonLink = ({
...
 			rel={opensInNewTab ? 'noreferrer noopener' : undefined}
 		>
-			{hasLeadingIcon && icon}
-			{hasText ? text : null}
-			{hasTrailingIcon && icon}
+			<>
+				{hasLeadingIcon && icon}
+				{hasText ? text : null}
+				{hasTrailingIcon && icon}
+			</>
```
Comment (BRKalow) [inline src/components/button-link/index.tsx:64]: The types were reporting that `Link` only accepts a single child, so I wrapped in a fragment here.

File: src/components/dev-dot-content/index.tsx [hunk 3]
```diff
@@ -88,6 +88,7 @@ const DevDotContent = ({
 		content = (
 			<MDXRemote
 				{...mdxRemoteProps}
+				// @ts-expect-error -- The next-mdx-remote types are wrong here, we don't want a record of ReactNode
```
Comment (BRKalow) [inline src/components/dev-dot-content/index.tsx:91]: We're on an old version of `next-mdx-remote` with incorrect types, so where I couldn't make updates in this repo I added this comment.

File: src/components/dev-dot-content/mdx-components/mdx-alert/types.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
+import { ReactElement } from 'react'
```
Comment (BRKalow) [inline src/components/dev-dot-content/mdx-components/mdx-alert/types.ts:6]:
```suggestion
import { type ReactElement } from 'react'
```

File: src/components/tabs/components/tab/types.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import { ReactNode } from 'react'
+import React, { ReactElement, ReactNode } from 'react'
```
Comment (BRKalow) [inline src/components/tabs/components/tab/types.ts:6]:
```suggestion
import { type ReactElement, type ReactNode } from 'react'
```

File: src/views/docs-view/types.ts [hunk 6]
```diff
@@ -23,12 +23,6 @@ export interface DocsViewProps {
 	 */
 	mdxSource: MDXRemoteSerializeResult
 
-	/**
-	 * Identical to the `MDXRemoteProps['lazy']` prop. The value is passed
-	 * unchanged to `MDXRemote`.
-	 */
-	lazy?: boolean
```
Comment (BRKalow) [inline src/views/docs-view/types.ts:30]: This prop is unused

File: src/views/product-integrations-landing/contexts/integrations-search-context.tsx [hunk 7]
```diff
@@ -10,7 +10,7 @@ import {
 	Tier,
 	IntegrationType,
 } from 'lib/integrations-api-client/integration'
-import { createContext, useContext, useMemo } from 'react'
+import { type ReactNode, createContext, useContext, useMemo } from 'react'
```
Comment (thiskevinwang) [inline src/views/product-integrations-landing/contexts/integrations-search-context.tsx:13]: Only if you need it... there's a `PropsWithChildren<T>` type that accepts an optional type argument

Comment (BRKalow) [reply src/views/product-integrations-landing/contexts/integrations-search-context.tsx:13]: Ahh word, good to know! 🙇

File: package.json [hunk 8]
```diff
@@ -114,15 +115,15 @@
 		"@hashicorp/react-alert-banner": "^7.0.5",
 		"@hashicorp/react-button": "^6.3.1",
 		"@hashicorp/react-code-block": "^6.3.0",
-		"@hashicorp/react-consent-manager": "^9.0.1",
-		"@hashicorp/react-docs-page": "^17.8.0",
+		"@hashicorp/react-consent-manager": "^9.1.1",
+		"@hashicorp/react-docs-page": "^17.9.0",
```
Comment (BRKalow) [inline package.json:119]: I bumped this to a version that was updated for react v18 types

File: package.json [hunk 9]
```diff
@@ -114,15 +115,15 @@
...
 		"@hashicorp/react-docs-sidenav": "^9.1.0",
 		"@hashicorp/react-enterprise-alert": "^7.0.1",
 		"@hashicorp/react-error-view": "^0.2.0",
 		"@hashicorp/react-hashi-stack-menu": "^3.0.0",
 		"@hashicorp/react-head": "^3.3.2",
 		"@hashicorp/react-inline-svg": "^6.1.0",
-		"@hashicorp/react-marketo-form": "^0.8.2",
+		"@hashicorp/react-marketo-form": "^0.8.4",
```
Comment (BRKalow) [inline package.json:126]: I bumped this to address a type error based on updated types for `window.analytics`

**Pull request merged**